### PR TITLE
SHS-5182:  Warbler button styles/utility class button styles update

### PR DESCRIPTION
**Pull request by cienvaras**
# READY FOR REVIEW

## Summary
- Update styles of `.hs-button` and `.hs-secondary-button` utility classes in warbler color scheme to ensure that they looks as expected in all cases
- Fix styles of table header links in warbler color scheme

## Need Review By (Date)
01/26

## Urgency
medium

## Steps to Test
1. Install the sociology site
2. Visit the [Traditional theme settings](https://sociology.suhumsci.loc/admin/appearance/settings/humsci_traditional) and change the color pairing to "Warbler"
3. Visit the ["PHD Placement Table" display of the "Sociology People" view config page](https://sociology.suhumsci.loc/admin/structure/views/view/custm_duplicate_of_default_people/edit/block_4), click on the  "Settings" link in the "Format" section and make the "PhD Year" column sortable.
    Screenshot 2024-01-24 at 17 00 58
4. Click on the "Content: Link to edit Content (Edit)" link in the "Fields" section, go to "Style Settings" and change the CSS class to `hs-button`
5. Save the view.
6. Visit the [PhD Placements page](https://sociology.suhumsci.loc/people/phd-placements) and confirm the following:
  a. The "PhD Year" label in the heading table is a link and its color is white
  b. The "edit" button text is white in all cases
    Screenshot 2024-01-24 at 17 06 52
7. Edit the [view config page](https://sociology.suhumsci.loc/admin/structure/views/view/custm_duplicate_of_default_people/edit/block_4) again and change back the "Content: Link to edit Content (Edit)" field class to `hs-secondary-button`. Go back to the [PhD Placements page](https://sociology.suhumsci.loc/people/phd-placements) and confirm that the "edit" button text is white on hover.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/components/_buttons.scss [hunk 1]
```diff
@@ -28,6 +28,12 @@ $buttons: (
     .hb-dark-inversion .views-element-container & {
       @include hb-global-color('color', 'white');
     }
+
+    @include hb-traditional {
+      .ht-pairing-warbler & {
+        color: white !important;
```
Comment (mndonx) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_buttons.scss:34]: I'm guessing  you had no other option but to use !important here... this site is so hard for overrides.

File: docroot/themes/humsci/humsci_basic/src/scss/components/_buttons.scss [hunk 2]
```diff
@@ -65,6 +71,12 @@ button.more-link,
   .hb-dark-inversion .views-element-container & {
     @include hb-global-color('color', 'white');
   }
+
+  @include hb-traditional {
+    .ht-pairing-warbler &:hover, &:focus {
```
Comment (mndonx) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_buttons.scss]: Maybe I'm misreading, but should this be: 

```suggestion
    .ht-pairing-warbler &:hover, 
    .ht-pairing-warbler &:focus {
```

Perhaps it compiles differently than I would expect?

**Review by mndonx — CHANGES_REQUESTED** [review]
@cienvaras Everything works as expected except for the focus state on the secondary button style - for me it is showing white text on white background. Otherwise, looks good!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_buttons.scss (hunk 1 above)
Comment (cienvaras) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_buttons.scss:34]: Yes, overrides in this case are hard, the clases used for setting the colors are really specific 🤷🏽‍♂️

**Comment by cienvaras** [comment]
@mndonx Thanks! It's fixed now, I thought that the focus styles matched the hoves ones, but that's not the case. It should work now.

**Pull request merged**